backend/src/utils/db.py

```
@@ -0,0 +1,166 @@
+            trade_data.get('type'),
+            trade_data.get('amount_sol'),
+            trade_data.get('amount_tokens'),
+            trade_data.get('price_usd'),
+            trade_data.get('transaction_id'),
+            trade_data.get('status'),
+            trade_data.get('timestamp', datetime.now().isoformat())
+        ))
+        conn.commit()
+    except sqlite3.IntegrityError:
+        pass # Already exists
+    except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)